### ISSUE TITLE
Failing to successfully connect to Remarkable 2

### ISSUE THREAD

**Issue by ju6ge** [naclander/Calibre-Remarkable-Device-Driver-Plugin#11]
Hey I installed your plugin for calibre and i can see that the connection with ssh is working, but after that the plugins fails to correctly setup the folders. Here is the calibre log :wink: 

```
PING vesemir.16.wlandt.de (10.128.16.81) 56(84) bytes of data.
64 bytes from vesemir.16.wlandt.de (10.128.16.81): icmp_seq=1 ttl=63 time=20.1 ms

--- vesemir.16.wlandt.de ping statistics ---
1 packets transmitted, 1 received, 0% packet loss, time 0ms
rtt min/avg/max/mdev = 20.148/20.148/20.148/0.000 ms
Trying to connect to vesemir
Devcie vesemir Present
Opening device
Traceback (most recent call last):
  File "/home/judge/.config/calibre/plugins/Remarkable Plugin.zip/remarkable_fs/documents.py", line 291, in load_node_without_linking
    node = cls(self, id, metadata)
  File "/home/judge/.config/calibre/plugins/Remarkable Plugin.zip/remarkable_fs/documents.py", line 364, in __init__
    self._size = self.root.sftp.stat(self.id + "." + self.file_type()).st_size
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 493, in stat
    t, msg = self._request(CMD_STAT, path)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 813, in _request
    return self._read_response(num)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 865, in _read_response
    self._convert_status(msg)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 894, in _convert_status
    raise IOError(errno.ENOENT, text)
FileNotFoundError: [Errno 2] No such file
Traceback (most recent call last):
  File "/home/judge/.config/calibre/plugins/Remarkable Plugin.zip/remarkable_fs/documents.py", line 291, in load_node_without_linking
    node = cls(self, id, metadata)
  File "/home/judge/.config/calibre/plugins/Remarkable Plugin.zip/remarkable_fs/documents.py", line 364, in __init__
    self._size = self.root.sftp.stat(self.id + "." + self.file_type()).st_size
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 493, in stat
    t, msg = self._request(CMD_STAT, path)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 813, in _request
    return self._read_response(num)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 865, in _read_response
    self._convert_status(msg)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 894, in _convert_status
    raise IOError(errno.ENOENT, text)
FileNotFoundError: [Errno 2] No such file
Traceback (most recent call last):
  File "/home/judge/.config/calibre/plugins/Remarkable Plugin.zip/remarkable_fs/documents.py", line 291, in load_node_without_linking
    node = cls(self, id, metadata)
  File "/home/judge/.config/calibre/plugins/Remarkable Plugin.zip/remarkable_fs/documents.py", line 364, in __init__
    self._size = self.root.sftp.stat(self.id + "." + self.file_type()).st_size
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 493, in stat
    t, msg = self._request(CMD_STAT, path)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 813, in _request
    return self._read_response(num)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 865, in _read_response
    self._convert_status(msg)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 894, in _convert_status
    raise IOError(errno.ENOENT, text)
FileNotFoundError: [Errno 2] No such file
Traceback (most recent call last):
  File "/home/judge/.config/calibre/plugins/Remarkable Plugin.zip/remarkable_fs/documents.py", line 291, in load_node_without_linking
    node = cls(self, id, metadata)
  File "/home/judge/.config/calibre/plugins/Remarkable Plugin.zip/remarkable_fs/documents.py", line 364, in __init__
    self._size = self.root.sftp.stat(self.id + "." + self.file_type()).st_size
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 493, in stat
    t, msg = self._request(CMD_STAT, path)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 813, in _request
    return self._read_response(num)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 865, in _read_response
    self._convert_status(msg)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 894, in _convert_status
    raise IOError(errno.ENOENT, text)
FileNotFoundError: [Errno 2] No such file
Traceback (most recent call last):
  File "/usr/lib/calibre/calibre/gui2/device.py", line 297, in detect_device
    dev.open(cd, self.current_library_uuid)
  File "calibre_plugins.remarkable_plugin.__init__", line 96, in open
    self.device_total_space = 1024 * int(stdout.read())
ValueError: invalid literal for int() with base 10: b''
Traceback (most recent call last):
  File "/home/judge/.config/calibre/plugins/Remarkable Plugin.zip/remarkable_fs/documents.py", line 291, in load_node_without_linking
    node = cls(self, id, metadata)
  File "/home/judge/.config/calibre/plugins/Remarkable Plugin.zip/remarkable_fs/documents.py", line 364, in __init__
    self._size = self.root.sftp.stat(self.id + "." + self.file_type()).st_size
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 493, in stat
    t, msg = self._request(CMD_STAT, path)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 813, in _request
    return self._read_response(num)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 865, in _read_response
    self._convert_status(msg)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 894, in _convert_status
    raise IOError(errno.ENOENT, text)
FileNotFoundError: [Errno 2] No such file
Traceback (most recent call last):
  File "/home/judge/.config/calibre/plugins/Remarkable Plugin.zip/remarkable_fs/documents.py", line 291, in load_node_without_linking
    node = cls(self, id, metadata)
  File "/home/judge/.config/calibre/plugins/Remarkable Plugin.zip/remarkable_fs/documents.py", line 364, in __init__
    self._size = self.root.sftp.stat(self.id + "." + self.file_type()).st_size
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 493, in stat
    t, msg = self._request(CMD_STAT, path)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 813, in _request
    return self._read_response(num)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 865, in _read_response
    self._convert_status(msg)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 894, in _convert_status
    raise IOError(errno.ENOENT, text)
FileNotFoundError: [Errno 2] No such file
Traceback (most recent call last):
  File "/home/judge/.config/calibre/plugins/Remarkable Plugin.zip/remarkable_fs/documents.py", line 291, in load_node_without_linking
    node = cls(self, id, metadata)
  File "/home/judge/.config/calibre/plugins/Remarkable Plugin.zip/remarkable_fs/documents.py", line 364, in __init__
    self._size = self.root.sftp.stat(self.id + "." + self.file_type()).st_size
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 493, in stat
    t, msg = self._request(CMD_STAT, path)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 813, in _request
    return self._read_response(num)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 865, in _read_response
    self._convert_status(msg)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 894, in _convert_status
    raise IOError(errno.ENOENT, text)
FileNotFoundError: [Errno 2] No such file
Traceback (most recent call last):
  File "/home/judge/.config/calibre/plugins/Remarkable Plugin.zip/remarkable_fs/documents.py", line 291, in load_node_without_linking
    node = cls(self, id, metadata)
  File "/home/judge/.config/calibre/plugins/Remarkable Plugin.zip/remarkable_fs/documents.py", line 364, in __init__
    self._size = self.root.sftp.stat(self.id + "." + self.file_type()).st_size
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 493, in stat
    t, msg = self._request(CMD_STAT, path)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 813, in _request
    return self._read_response(num)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 865, in _read_response
    self._convert_status(msg)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 894, in _convert_status
    raise IOError(errno.ENOENT, text)
FileNotFoundError: [Errno 2] No such file
Traceback (most recent call last):
  File "/usr/lib/calibre/calibre/gui2/device.py", line 297, in detect_device
    dev.open(cd, self.current_library_uuid)
  File "calibre_plugins.remarkable_plugin.__init__", line 96, in open
    self.device_total_space = 1024 * int(stdout.read())
ValueError: invalid literal for int() with base 10: b''
Traceback (most recent call last):
  File "/home/judge/.config/calibre/plugins/Remarkable Plugin.zip/remarkable_fs/documents.py", line 291, in load_node_without_linking
    node = cls(self, id, metadata)
  File "/home/judge/.config/calibre/plugins/Remarkable Plugin.zip/remarkable_fs/documents.py", line 364, in __init__
    self._size = self.root.sftp.stat(self.id + "." + self.file_type()).st_size
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 493, in stat
    t, msg = self._request(CMD_STAT, path)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 813, in _request
    return self._read_response(num)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 865, in _read_response
    self._convert_status(msg)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 894, in _convert_status
    raise IOError(errno.ENOENT, text)
FileNotFoundError: [Errno 2] No such file
Traceback (most recent call last):
  File "/home/judge/.config/calibre/plugins/Remarkable Plugin.zip/remarkable_fs/documents.py", line 291, in load_node_without_linking
    node = cls(self, id, metadata)
  File "/home/judge/.config/calibre/plugins/Remarkable Plugin.zip/remarkable_fs/documents.py", line 364, in __init__
    self._size = self.root.sftp.stat(self.id + "." + self.file_type()).st_size
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 493, in stat
    t, msg = self._request(CMD_STAT, path)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 813, in _request
    return self._read_response(num)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 865, in _read_response
    self._convert_status(msg)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 894, in _convert_status
    raise IOError(errno.ENOENT, text)
FileNotFoundError: [Errno 2] No such file
Traceback (most recent call last):
  File "/home/judge/.config/calibre/plugins/Remarkable Plugin.zip/remarkable_fs/documents.py", line 291, in load_node_without_linking
    node = cls(self, id, metadata)
  File "/home/judge/.config/calibre/plugins/Remarkable Plugin.zip/remarkable_fs/documents.py", line 364, in __init__
    self._size = self.root.sftp.stat(self.id + "." + self.file_type()).st_size
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 493, in stat
    t, msg = self._request(CMD_STAT, path)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 813, in _request
    return self._read_response(num)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 865, in _read_response
    self._convert_status(msg)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 894, in _convert_status
    raise IOError(errno.ENOENT, text)
FileNotFoundError: [Errno 2] No such file
Traceback (most recent call last):
  File "/home/judge/.config/calibre/plugins/Remarkable Plugin.zip/remarkable_fs/documents.py", line 291, in load_node_without_linking
    node = cls(self, id, metadata)
  File "/home/judge/.config/calibre/plugins/Remarkable Plugin.zip/remarkable_fs/documents.py", line 364, in __init__
    self._size = self.root.sftp.stat(self.id + "." + self.file_type()).st_size
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 493, in stat
    t, msg = self._request(CMD_STAT, path)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 813, in _request
    return self._read_response(num)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 865, in _read_response
    self._convert_status(msg)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 894, in _convert_status
    raise IOError(errno.ENOENT, text)
FileNotFoundError: [Errno 2] No such file
Traceback (most recent call last):
  File "/usr/lib/calibre/calibre/gui2/device.py", line 297, in detect_device
    dev.open(cd, self.current_library_uuid)
  File "calibre_plugins.remarkable_plugin.__init__", line 96, in open
    self.device_total_space = 1024 * int(stdout.read())
ValueError: invalid literal for int() with base 10: b''
Traceback (most recent call last):
  File "/home/judge/.config/calibre/plugins/Remarkable Plugin.zip/remarkable_fs/documents.py", line 291, in load_node_without_linking
    node = cls(self, id, metadata)
  File "/home/judge/.config/calibre/plugins/Remarkable Plugin.zip/remarkable_fs/documents.py", line 364, in __init__
    self._size = self.root.sftp.stat(self.id + "." + self.file_type()).st_size
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 493, in stat
    t, msg = self._request(CMD_STAT, path)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 813, in _request
    return self._read_response(num)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 865, in _read_response
    self._convert_status(msg)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 894, in _convert_status
    raise IOError(errno.ENOENT, text)
FileNotFoundError: [Errno 2] No such file
Traceback (most recent call last):
  File "/home/judge/.config/calibre/plugins/Remarkable Plugin.zip/remarkable_fs/documents.py", line 291, in load_node_without_linking
    node = cls(self, id, metadata)
  File "/home/judge/.config/calibre/plugins/Remarkable Plugin.zip/remarkable_fs/documents.py", line 364, in __init__
    self._size = self.root.sftp.stat(self.id + "." + self.file_type()).st_size
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 493, in stat
    t, msg = self._request(CMD_STAT, path)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 813, in _request
    return self._read_response(num)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 865, in _read_response
    self._convert_status(msg)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 894, in _convert_status
    raise IOError(errno.ENOENT, text)
FileNotFoundError: [Errno 2] No such file
Traceback (most recent call last):
  File "/home/judge/.config/calibre/plugins/Remarkable Plugin.zip/remarkable_fs/documents.py", line 291, in load_node_without_linking
    node = cls(self, id, metadata)
  File "/home/judge/.config/calibre/plugins/Remarkable Plugin.zip/remarkable_fs/documents.py", line 364, in __init__
    self._size = self.root.sftp.stat(self.id + "." + self.file_type()).st_size
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 493, in stat
    t, msg = self._request(CMD_STAT, path)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 813, in _request
    return self._read_response(num)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 865, in _read_response
    self._convert_status(msg)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 894, in _convert_status
    raise IOError(errno.ENOENT, text)
FileNotFoundError: [Errno 2] No such file
Traceback (most recent call last):
  File "/home/judge/.config/calibre/plugins/Remarkable Plugin.zip/remarkable_fs/documents.py", line 291, in load_node_without_linking
    node = cls(self, id, metadata)
  File "/home/judge/.config/calibre/plugins/Remarkable Plugin.zip/remarkable_fs/documents.py", line 364, in __init__
    self._size = self.root.sftp.stat(self.id + "." + self.file_type()).st_size
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 493, in stat
    t, msg = self._request(CMD_STAT, path)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 813, in _request
    return self._read_response(num)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 865, in _read_response
    self._convert_status(msg)
  File "/usr/lib/python3.10/site-packages/paramiko/sftp_client.py", line 894, in _convert_status
    raise IOError(errno.ENOENT, text)
FileNotFoundError: [Errno 2] No such file
Traceback (most recent call last):
  File "/usr/lib/calibre/calibre/gui2/device.py", line 297, in detect_device
    dev.open(cd, self.current_library_uuid)
  File "calibre_plugins.remarkable_plugin.__init__", line 96, in open
    self.device_total_space = 1024 * int(stdout.read())
ValueError: invalid literal for int() with base 10: b''
Creating document root
Finished creating document root
Error while trying to open True (Driver: <calibre_plugins.remarkable_plugin.RemarkablePlugin object at 0x7f2c607ab9d0>)
Opening device
Creating document root
Finished creating document root
Error while trying to open True (Driver: <calibre_plugins.remarkable_plugin.RemarkablePlugin object at 0x7f2c607ab9d0>)
Opening device
Creating document root
Finished creating document root
Error while trying to open True (Driver: <calibre_plugins.remarkable_plugin.RemarkablePlugin object at 0x7f2c607ab9d0>)
Opening device
Creating document root
Finished creating document root
Error while trying to open True (Driver: <calibre_plugins.remarkable_plugin.RemarkablePlugin object at 0x7f2c607ab9d0>)

```

**Comment by jonaustin** [comment]
Had the same issue; your block device is probably different from the one hard-coded in this plugin: https://github.com/naclander/Calibre-Remarkable-Device-Driver-Plugin/pull/15

Note: the `FileNotFound` errors are a red herring; those seem to just occur for any Remarkable Notebook files on mine anyway (i.e. I printed out the metadata and they're all just notebooks).

**Comment by naclander** [comment]
Hey @ju6ge  I've merged #15, you can try the latest code from master or wait until I cut a new release, but hopefully this fixes the immediate issue.

**Comment by ju6ge** [comment]
Currently I am swamped with work, but I will take a look as soon as I have some time ;)
